simulator/simulator.cpp

```diff
@@ -38,8 +39,16 @@ class SimulatorFactory {
     template<typename T, Endian e>
     struct TBuilder : public Builder {
         TBuilder() = default;
-        std::unique_ptr<Simulator> get_funcsim( bool log) final { return std::make_unique<FuncSim<T>>( e, log); }
-        std::unique_ptr<CycleAccurateSimulator> get_perfsim() final { return std::make_unique<PerfSim<T>>( e); }
+        std::unique_ptr<Simulator> get_funcsim( bool log) final {
+            auto func_sim = std::make_unique<FuncSim<T>>( e, log);
+            func_sim->set_isa( Builder::isa);
```

Comment:
Can we pass it to ctor directly? It does not look very safe to leave `set_isa` accessible for everyone at every moment.
By the way, later we may define endian based on string to reduce # of ctor arguments.